src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/string_view.hpp

```diff
@@ -14,92 +14,489 @@
+        if (bsoncxx::detail::distance(found) != static_cast<difference_type>(infix.size())) {
+            return npos;
+        }
+        return static_cast<size_type>(rend() - found.end());
+    }
+
+    /**
+     * @brief Find the zero-based index of the left-most occurrence of any character of the given
+     * set, starting at pos
+     */
+    constexpr size_type find_first_of(self_type set, size_type pos = 0) const noexcept {
+        return _find_if(pos, bsoncxx::detail::equal_to_any_of(set));
+    }
+
+    /**
+     * @brief Find the zero-based index of the right-most occurrence of any character of the
+     * given set, starting at (and including) pos
+     */
+    constexpr size_type find_last_of(self_type set, size_type pos = npos) const noexcept {
+        return _rfind_if(pos, bsoncxx::detail::equal_to_any_of(set));
+    }
+
+    /**
+     * @brief Find the zero-based index of the left-most occurrence of any character that
+     * is NOT a member of the given set of characters
+     */
+    constexpr size_type find_first_not_of(self_type set, size_type pos = 0) const noexcept {
+        return _find_if(pos, bsoncxx::detail::not_fn(bsoncxx::detail::equal_to_any_of(set)));
+    }
+
+    /**
+     * @brief Find the zero-based index of the right-most occurrence of any character that
+     * is NOT a member of the given set of characters, starting at (and including) pos
+     */
+    constexpr size_type find_last_not_of(self_type set, size_type pos = npos) const noexcept {
+        return _rfind_if(pos, bsoncxx::detail::not_fn(bsoncxx::detail::equal_to_any_of(set)));
+    }
+
+#pragma push_macro("DECL_FINDERS")
+#undef DECL_FINDERS
+#define DECL_FINDERS(Name, DefaultPos)                                                            \
+    constexpr size_type Name(value_type chr, size_type pos = DefaultPos) const noexcept {         \
+        return Name(self_type(&chr, 1), pos);                                                     \
+    }                                                                                             \
+    constexpr size_type Name(const_pointer cstr, size_type pos, size_type count) const noexcept { \
+        return Name(self_type(cstr, count), pos);                                                 \
+    }                                                                                             \
+    constexpr size_type Name(const_pointer cstr, size_type pos = DefaultPos) const noexcept {     \
+        return Name(self_type(cstr), pos);                                                        \
+    }                                                                                             \
+    BSONCXX_FORCE_SEMICOLON
+    DECL_FINDERS(find, 0);
+    DECL_FINDERS(rfind, npos);
+    DECL_FINDERS(find_first_of, 0);
+    DECL_FINDERS(find_last_of, npos);
+    DECL_FINDERS(find_first_not_of, 0);
+    DECL_FINDERS(find_last_not_of, npos);
+#pragma pop_macro("DECL_FINDERS")
```

Comment:
Macro guards in config headers preceed the `pop_macro` with an `#undef`:

```cpp
#pragma push_macro("X")
#undef X
#define X
...
#undef X
#pragma pop_macro("X")
```

Recommend including the `#undef` here as well for consistency. Also applies to other instances of macro guards in this PR.

Reply:
Does that `undef` really do anything if the macro definition is immediately popped?

Reply:
On second thought, probably not. I thought it would trigger a warning if the macro being undef'd was not already defined (e.g. as a typo check), but that does not seem to be the case for either GCC or Clang in my local tests. The `pop_macro` doesn't seem to generate any warnings if the macro being popped is already defined either. The `#undef` preceeding a `#pragma pop_macro` doesn't seem to serve a useful function, so I'm in favor of it's omission going forward.